### DISCUSSION TITLE
Tested, working fine with PC5010 (aka DSC Power 832)

### DISCUSSION THREAD

**Discussion by zzhang1** [taligentx/dscKeybusInterface#123] [dscKeybusInterface Discussion]
Hi! Almost one year in, just wanted to report belatedly that everything works great on my PC5010 system. It is a single partition with a 24 of 32 zones in use. 

Working fine:
- Arming home/away and disarming
- Window sensors
- Door sensors
- Motion sensors
- Smoke alarms
- System trouble flag

Untested:
- Low temperature sensor

**Comment by jmlzone** [comment]
I just started working with PC5010.  Also works fine.  Tried a little to see if I could get the zone tables from the lcd5500Z keypads, but could not make sense of the data when I told a keypad to broadcast the labels.

Additional to the list above, tried output 3 and output 4 from my expansion box, working fine.
my system has single partition, 2 LCD5500Z keypads, A 433 keypad, expansion box with a second power supply and PC5400 printer interface (I never configured it) expansion zone and outputs connected to other things in the house.

**Comment by taligentx** [comment]
@zzhang1 @jmlzone Thanks for confirming the interface works with the PC5010!  I've updated the list of verified panels.

**Comment by jmlzone** [comment]
Thanks for the easy plug and play code.  I will need o get back to my home control project that gies through the pc5010.  I have some expansion GPIO that I use to open/close my garage doors and I was also able to use the interface to actuate those!

**Reply by crlesor** [reply]
@zzhang1 @jmlzone I am starting to connect my PC5010 using this code and my Arduino mega 2560 with ethernet shield.  When I run both the KeybusReader and Status codes to see the signal from the PC5010, the message says  "DSC Keybus Interface is online.
Keybus connected
Keybus disconnected".  Did you wire the PC5010 as a classic series?

**Reply by kricon** [reply]
Hi @crlesor, which version of PC5010 you have? V1 uses 40pin main IC with 06000464 written on it, while V2 and up uses 28pin IC with written version on main chip (or 06000666 for early revisions). There also could be some region-based differences with different models, revisions and chips.
If you have 40pin chip (V1/06000664), that means first generation of PowerSeries which assign zones in group of 4 (instead of 8 for all newer versions) and maybe that is what's confusing it from using with this library.

It could be also bad wiring (especially if using dupont wires or breadboard - from my experience connection is better just twisting the wires onto resistors/transistor than using cheap breadboard and dupoint wires). What resistors did you use?

You're supposed to wire/define it as a PowerSeries.

**Comment by crlesor** [comment]
Hi @kricon, thank you for such a detailed response.  Attached is a picture of my control panel.  I do have a chip with 28 pins but it has different text printed on it.  Below the 28 pin, there is another chip with a similar number (06000309) to the two that you mentioned.

I have tried wiring my board with both the classic and non classic configuration and get the same serial output.

Does this picture help to diagnose the problem that I am seeing?

**Reply by crlesor** [reply]
@kricon it looks like my picture was not included in my last message.
![PXL_20230517_160533967](https://github.com/taligentx/dscKeybusInterface/assets/61602340/18a828ca-d146-4db1-95d0-8985a30c750e)

**Reply by kricon** [reply]
@crlesor Hi, the version is indicated after letter V on main chip (V2.02). I've tested on same panel version, working fine.
Again, you should **not** use Classic series define with this panel. Which resistors do you use and how you wired them to the Mega?

**Reply by crlesor** [reply]
@kricon I commented out the #define dscClassicSeries line in both the KeybusReader and Status sketches.  I still get the disconnected message in the serial monitor.  I am using a breadboard but only to confirm that I can read something in the serial monitor.  I checked the resistance from the green and yellow terminals on the control panel and the pin of the dupont wire that plugs into the Arduino and instead of the 15Kohm that I measure directly across the resister, the entire circuit is 20Kohm.  Would the extra 5Kohm be enough to prevent the Arduino from reading anything from the control panel?

I also disconnected the write circuit from the Arduino in case the 2N3904 transistor was causing problems.  Still do not get any readings from the control panel in the Arduino serial monitor.
